pymeasure/instruments/sorensen/sorensen30035E.py

```diff
@@ -0,0 +1,173 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_range
+
+import time
+
+
+class Sorensen30035E(Instrument):
+    """ Represents the Sorensen300-3.5E power supply with GPIB option and provides a
+    high-level interface for interacting with the instrument. Use good judgement and
+    protect the power supply outputs with flyback and blocking diodes if you have an
+    inductive load. See the Sorensen manual for advice.
+
+
+    .. code-block:: python
+
+        sorensen = Sorensen30035E("GPIB::1")
+
+        sorensen.enable_source()                # Enables output of current
+        sorensen.ramp_to_voltage(200,200)       # Ramp to 200 V in 200 seconds nominally
+
+        print(sorensen.current)                 # Prints the actual current output in Amps
+
+        sorensen.shutdown()                     # Ramps the current to 0 mA and disables output
+
+    """
+
+    def __init__(self, adapter, **kwargs):
+        super(Sorensen30035E, self).__init__(
+            adapter, "Sorensen 300-3.5E DC Power Supply", **kwargs
+        )
+
+    output_delay = Instrument.control(
+        "OUTPut:PROTection:DELay?", "OUTPut:PROTection:DELay %g",
+        """ A floating point property that sets a manual delay for the source
+        after the output is turned on before a measurement is taken and the reverse.
+        between 0 [seconds] and 999.9999 [seconds].""",
```

Comment:
Some general comments regarding docstrings (which you already might know, as it is an old implementation) (see also our [documentation](https://pymeasure.readthedocs.io/en/latest/dev/coding_standards.html#docstrings)): They should be in imperative form, i.e. "Control" instead of "Controls".

We also changed, how we want to write docstrings for properties (the same [documentation link](https://pymeasure.readthedocs.io/en/latest/dev/coding_standards.html#docstrings), and [here examples](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments.html#adding-instruments)): They should start with "Control", "Measure", or "Set", according to the property type.

If you have time, it would be nice to refactor the docstrings.